### PR TITLE
DOC: Fix quantile docstring

### LINKED ISSUE #22898: DOC: Fix the docstring of quantile in pandas/core/frame.py
pandas docstring are expected to follow the format defined in https://pandas.pydata.org/pandas-docs/stable/contributing_docstring.html

This includes among other things:
- There is a short description fitting in one line (there can be a longer one after that).
- The description of the parameters, the returns section, and the see also items, starts with a capital letter and ends with a period.
- In the Return section there is no name before the type (e.g. `transformed : Series`) but just the type (e.g. `Series`) in the first line.
- The examples are valid PEP-8 code, and when executed, they produce the shown input.

We should make the required adjustments in the docstring of the function `quantile` in `pandas/core/frame.py`.

We should validate the docstring with our validator (which does not validate all the formats, but some):
- `./scripts/validate_docstrings.py pandas.DataFrame.quantile`

And we should validate the PEP-8 of the examples (note that the next command will validate the PEP-8 of all examples, just make sure that none of the shown is in the function being addressed, the rest will be fixed in separate issues):
- `flake8 --doctests pandas/core/frame.py`

### PR DESCRIPTION
**Pull request by tm9k1**
…tput of example 1 Added summary for **See Also** functions Some typo fixes .

- [x] closes #22898
- [ ] tests added / passed
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

**Comment by brute4s99** [comment]
In this commit -
Added extended summary for the function
corrected output of example 1
Added summary for **See Also** functions
Some typo fixes

Comments :
Also, please review my implementation to use '\' for multi-line commands. `./scripts/validate_docstrings.py` likes it ! :D

File: pandas/core/frame.py [hunk 1]
```diff
@@ -7416,23 +7419,22 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
               index is the columns of self and the values are the quantiles.
 
         Examples
         --------
-
-        >>> df = pd.DataFrame(np.array([[1, 1], [2, 10], [3, 100], [4, 100]]),
-                              columns=['a', 'b'])
+        >>> df = pd.DataFrame(np.array([[1, 1], [2, 10], \
+                                        [3, 100], [4, 100]]), \
+                                        columns=['a', 'b'])
```
Comment (datapythonista) [inline pandas/core/frame.py]: We usually try to keep the examples small, but I think the results are almost meaningless with just two samples. I'd have may be 6 or 8 in this case. Also, we try to use data that looks kind of real, and that the user know before hand (see this example: https://github.com/pandas-dev/pandas/blob/master/pandas/core/frame.py#L7580).

Also, can you use the validations mentioned in the issue. This example can't be run, and the backslashes are redundant and a style error is probably being shown.

File: pandas/core/frame.py [hunk 2]
```diff
@@ -7441,11 +7443,12 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
-        >>> df = pd.DataFrame({'A': [1, 2],
-                               'B': [pd.Timestamp('2010'),
-                                     pd.Timestamp('2011')],
-                               'C': [pd.Timedelta('1 days'),
-                                     pd.Timedelta('2 days')]})
+        >>> df = pd.DataFrame({ 'A': [1, 2], \
+                                'B': [pd.Timestamp('2010'), \
+                                      pd.Timestamp('2011')], \
+                                'C': [pd.Timedelta('1 days'), \
+                                      pd.Timedelta('2 days')]})
```
Comment (datapythonista) [inline pandas/core/frame.py]: If possible, try to define just an example at the beginning, and use it everywhere. If you need a date column, you can add it there.

**Review by datapythonista — COMMENTED** [review]
Looking good, thanks for the fixes. I added some comments on things that can be improved.

File: pandas/core/frame.py [hunk 3]
```diff
@@ -7390,20 +7390,23 @@ def f(s):
...
         ----------
-        q : float or array-like, default 0.5 (50% quantile)
-            0 <= q <= 1, the quantile(s) to compute
-        axis : {0, 1, 'index', 'columns'} (default 0)
-            0 or 'index' for row-wise, 1 or 'columns' for column-wise
+        q : float or array-like, default 0.5 (50% quantile) [0 <= q <= 1]
+            The quantile(s) to compute.
+        axis : boolean{0, 1, 'index', 'columns'} (default 0)
+            For row-wise : 0 or'index', for column-wise : 1 or 'columns'.
         numeric_only : boolean, default True
```
[the remark this answers is not in the document]
Comment (dquore) [reply pandas/core/frame.py]: I'll switch to the convention `axis : {0 or 'index', 1 or 'columns'}, default 0` so not boolean/bool won't be needed.

File: pandas/core/frame.py [hunk 4]
```diff
@@ -7390,20 +7390,23 @@ def f(s):
...
 
         Parameters
         ----------
-        q : float or array-like, default 0.5 (50% quantile)
-            0 <= q <= 1, the quantile(s) to compute
-        axis : {0, 1, 'index', 'columns'} (default 0)
-            0 or 'index' for row-wise, 1 or 'columns' for column-wise
+        q : float or array-like, default 0.5 (50% quantile) [0 <= q <= 1]
+            The quantile(s) to compute.
+        axis : boolean{0, 1, 'index', 'columns'} (default 0)
```
[the remark this answers is not in the document]
Comment (dquore) [reply pandas/core/frame.py]: switching to convention then, please wait for next commit.

File: pandas/core/frame.py [hunk 5]
```diff
@@ -7390,20 +7390,23 @@ def f(s):
...
+        In case of insufficient number of data points for clean division
+        into groups, specify `interpolation` scheme to implement.
 
         Parameters
         ----------
-        q : float or array-like, default 0.5 (50% quantile)
-            0 <= q <= 1, the quantile(s) to compute
-        axis : {0, 1, 'index', 'columns'} (default 0)
-            0 or 'index' for row-wise, 1 or 'columns' for column-wise
+        q : float or array-like, default 0.5 (50% quantile) [0 <= q <= 1]
```
[the remark this answers is not in the document]
Comment (dquore) [reply pandas/core/frame.py]: Sure, @datapythonista ! Looks much better that way!

File: pandas/core/frame.py [hunk 6]
```diff
@@ -7390,20 +7390,23 @@ def f(s):
     def quantile(self, q=0.5, axis=0, numeric_only=True,
                  interpolation='linear'):
         """
-        Return values at the given quantile over requested axis.
+        Return value(s) at the given quantile over requested axis.
+
+        This function calculates the 'q' quantile values on the dataframe,
```
[the remark this answers is not in the document]
Comment (dquore) [reply pandas/core/frame.py]: Typos fixed, and changed terminology. Wait for next commit!

File: pandas/core/frame.py [hunk 7]
```diff
@@ -7416,23 +7419,22 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
 
         Returns
         -------
-        quantiles : Series or DataFrame
-
-            - If ``q`` is an array, a DataFrame will be returned where the
-              index is ``q``, the columns are the columns of self, and the
+        scalar, Series or DataFrame
```
[the remark this answers is not in the document]
Comment (dquore) [reply pandas/core/frame.py]: Sure, I'll add a few examples !

File: pandas/core/frame.py [hunk 8]
```diff
@@ -7455,7 +7458,9 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
         See Also
         --------
         pandas.core.window.Rolling.quantile
```
[the remark this answers is not in the document]
Comment (dquore) [reply pandas/core/frame.py]: Alright, @datapythonista !

File: pandas/core/frame.py (hunk 1 above)
Comment (dquore) [reply pandas/core/frame.py]: Alright, @datapythonista . I'll think of something!

File: pandas/core/frame.py (hunk 2 above)
Comment (dquore) [reply pandas/core/frame.py]: okay, @datapythonista .

File: pandas/core/frame.py (hunk 6 above)
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: Typos fixed, and changed terminology. Wait for next commit!

File: pandas/core/frame.py (hunk 5 above)
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: Sure, @datapythonista ! Looks much better that way!

File: pandas/core/frame.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: I'll switch to the convention axis : {0 or 'index', 1 or 'columns'}, default 0 so boolean/bool won't be needed.

File: pandas/core/frame.py (hunk 7 above)
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: Sure, I'll add a few examples !

File: pandas/core/frame.py (hunk 8 above)
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: Alright, @datapythonista !

File: pandas/core/frame.py (hunk 1 above)
Comment (brute4s99) [reply pandas/core/frame.py]: Alright, @datapythonista . I'll think of something!

File: pandas/core/frame.py (hunk 2 above)
Comment (brute4s99) [reply pandas/core/frame.py]: okay, @datapythonista .

File: pandas/core/frame.py (hunk 1 above)
Comment (brute4s99) [reply pandas/core/frame.py]: @datapythonista please review my PR. I believe these examples should suffice.

File: pandas/core/frame.py [hunk 9]
```diff
@@ -7416,46 +7422,69 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
+        2    423
+        3    859
+        4     32
+        5    548
+        6     33
+        7    951
+        8    450
+        9   1001
+        10   998
+        >>> for i in sorted(df['Data'],reverse=True): print(i)
```
Comment (datapythonista) [inline pandas/core/frame.py]: The preferred way to sort a Series is `df['Data'].sort_values()` which also have a `reverse` parameter

File: pandas/core/frame.py [hunk 10]
```diff
@@ -7416,46 +7422,69 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
+        450
+        423
+        416
+        33
+        32
+        >>> df.quantile()
+        Data    493.0
+        Name: 0.5, dtype: float64
+        >>> type(df.quantile())
+        <class 'pandas.core.series.Series'>
```
Comment (datapythonista) [inline pandas/core/frame.py]: I don't think it's necessary to show the type

File: pandas/core/frame.py [hunk 11]
```diff
@@ -7416,46 +7422,69 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
+        >>> df.quantile(q=0.7)
+        Data    859.0
+        Name: 0.7, dtype: float64
+        >>> df.quantile(q=[0.5, 0.7])
```
Comment (datapythonista) [inline pandas/core/frame.py]: I'd use for `q` the values `0.05` and `0.95` as they are quite standard in practice

File: pandas/core/frame.py [hunk 12]
```diff
@@ -7416,46 +7422,69 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
+              Data
+        0.5  493.0
+        0.7  859.0
+        >>> df.quantile(q=[0.55],interpolation='higher')
+              Data
+        0.55   548
+        >>> df.quantile(q=[0.55],interpolation='lower')
```
Comment (datapythonista) [inline pandas/core/frame.py]: make sure you don't have pep8 issues in the code, a space is missing after the comma

**Review by datapythonista — CHANGES_REQUESTED** [review]
It's looking good, but there are some things that could make the examples better I think, added some comments

File: pandas/core/frame.py (hunk 9 above)
Comment (brute4s99) [reply pandas/core/frame.py]: Should've looked the documentation for this. Will correct this, @datapythonista .

File: pandas/core/frame.py [hunk 13]
```diff
@@ -7416,46 +7422,69 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
         See Also
         --------
         pandas.core.window.Rolling.quantile
+            Returns the rolling quantile for the DataFrame.
         numpy.percentile
+            Returns 'nth' percentile for the DataFrame.
+
+        Examples
+        --------
+        >>> import pandas as pd
```
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: sure!

File: pandas/core/frame.py [hunk 14]
```diff
@@ -7390,20 +7390,26 @@ def f(s):
...
         ----------
-        q : float or array-like, default 0.5 (50% quantile)
-            0 <= q <= 1, the quantile(s) to compute
-        axis : {0, 1, 'index', 'columns'} (default 0)
-            0 or 'index' for row-wise, 1 or 'columns' for column-wise
-        numeric_only : boolean, default True
+        q : float or array-like, default 0.5
+            The quantile(s) to compute (0 <= q <= 1) (0.5 == 50% quantile)
+            If float is passed as `q`, scalar quantile is returned
+            If `array-like` is passed as `q`, Series is returned.
```
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: woops! Missed it. Sorry!

[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: >q : float or array-like, default 0.5
>           The quantile(s) to compute
>            the quantile(s) should be a floating point number
>            in the range [0.0, 1.0]
>            Passing `q` = 0.5 is equivalent to call for a 50% quantile value 

Does this look fine, @datapythonista ?

File: pandas/core/frame.py [hunk 15]
```diff
@@ -7416,46 +7422,69 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
-        B    2010-07-02 12:00:00
-        C        1 days 12:00:00
-        Name: 0.5, dtype: object
-
         See Also
         --------
         pandas.core.window.Rolling.quantile
+            Returns the rolling quantile for the DataFrame.
         numpy.percentile
+            Returns 'nth' percentile for the DataFrame.
```
[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: corrected !

File: pandas/core/frame.py (hunk 14 above)
[the remark this answers is not in the document]
Comment (datapythonista) [reply pandas/core/frame.py]: Up to you, but I'd prefer `should be a float between 0 and 1 (inclusive).` or something like this, that doesn't seem a mix of text and mathematical notation. Also include periods to separate sentences... And you can mention that 50% quantile is the median.

[the remark this answers is not in the document]
Comment (brute4s99) [reply pandas/core/frame.py]: >... And you can mention that 50% quantile is the median.

I am having second thoughts about this, but, as you say ! :+1:

**Comment by jreback** [comment]
@brute4s99 can you update to @datapythonista comments

**Comment by brute4s99** [comment]
It's done, @jreback. I am awaiting a review now.
Apologies for being late, I was busy in PyCon India 2018

**Comment by brute4s99** [comment]
@datapythonista please review

File: pandas/core/frame.py [hunk 16]
```diff
@@ -7390,20 +7390,26 @@ def f(s):
...
-            0 <= q <= 1, the quantile(s) to compute
-        axis : {0, 1, 'index', 'columns'} (default 0)
-            0 or 'index' for row-wise, 1 or 'columns' for column-wise
-        numeric_only : boolean, default True
+        q : float or array-like, default 0.5
+            The quantile(s) to compute,
+            should be a float between 0 and 1 (inclusive),
+            0.5 is equivalent to calculate 50% quantile value ie the median.
+        axis : {0 or 'index', 1 or 'columns'}, default 0
+            For row-wise : 0 or'index', for column-wise : 1 or 'columns'.
```
Comment (datapythonista) [inline pandas/core/frame.py]: there is a typo here, but anyway this seems redundant with the type line. Can you check other docstrings with the `axis` parameter and copy their description.

File: pandas/core/frame.py [hunk 17]
```diff
@@ -7390,20 +7390,26 @@ def f(s):
...
+        into groups, specify `interpolation` scheme to implement.
 
         Parameters
         ----------
-        q : float or array-like, default 0.5 (50% quantile)
-            0 <= q <= 1, the quantile(s) to compute
-        axis : {0, 1, 'index', 'columns'} (default 0)
-            0 or 'index' for row-wise, 1 or 'columns' for column-wise
-        numeric_only : boolean, default True
+        q : float or array-like, default 0.5
```
Comment (datapythonista) [inline pandas/core/frame.py]: `array-like` is used for an object that implements the `numpy.array` api. I think just `list` would be better here.

File: pandas/core/frame.py [hunk 18]
```diff
@@ -7390,20 +7390,26 @@ def f(s):
...
+            The quantile(s) to compute,
+            should be a float between 0 and 1 (inclusive),
+            0.5 is equivalent to calculate 50% quantile value ie the median.
```
Comment (datapythonista) [inline pandas/core/frame.py]: Can you finish the lines to close to 79 characters long. Feels a bit weird to have these short lines.

Can you use `(i.e. the median)` at the end.

File: pandas/core/frame.py [hunk 19]
```diff
@@ -7416,46 +7422,58 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
-        A                    1.5
-        B    2010-07-02 12:00:00
-        C        1 days 12:00:00
-        Name: 0.5, dtype: object
+        Series
 
         See Also
         --------
         pandas.core.window.Rolling.quantile
+            Returns the rolling quantile for the DataFrame.
```
Comment (datapythonista) [inline pandas/core/frame.py]: I think the `See Also` section has the format `func : desc` in the same line (continuing in the next). Can you check the docs and the other docstrings and adapt.

File: pandas/core/frame.py [hunk 20]
```diff
@@ -7416,46 +7422,58 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
         numpy.percentile
+            Returns 'nth' percentile for numpy arrays.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> d = {'animal':['Cheetah','Falcon','Eagle','Goose','Pigeon'],
+        ... 'class':['mammal','bird','bird','bird','bird'],
+        ... 'max_speed':[120,np.nan,320,142,150]}
```
Comment (datapythonista) [inline pandas/core/frame.py]: please respect pep8 and have the right indentation. It'd be better if you can avoid having a separate variable `d`, and instead provide the data directly to the constructor.

The `import numpy` can be ommited.

File: pandas/core/frame.py [hunk 21]
```diff
@@ -7416,46 +7422,58 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
+        >>> df = pd.DataFrame(data=d)
+        >>> df
+            animal   class  max_speed
+        0  Cheetah  mammal        120.0
+        1   Falcon    bird        NaN
+        2    Eagle    bird        320.0
+        3    Goose    bird        142.0
+        4   Pigeon    bird        150.0
+
+        The `max_speed` in sorted order:-
```
Comment (datapythonista) [inline pandas/core/frame.py]: any reason for the `-` at the end or is a typo?

File: pandas/core/frame.py [hunk 22]
```diff
@@ -7416,46 +7422,58 @@ def quantile(self, q=0.5, axis=0, numeric_only=True,
...
         --------
         pandas.core.window.Rolling.quantile
+            Returns the rolling quantile for the DataFrame.
         numpy.percentile
+            Returns 'nth' percentile for numpy arrays.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> d = {'animal':['Cheetah','Falcon','Eagle','Goose','Pigeon'],
```
Comment (datapythonista) [inline pandas/core/frame.py]: I'd have the name of the animals as the index instead.

**Review by datapythonista — CHANGES_REQUESTED** [review]
Added many comments, but there are several other pep8 problems. As I mention in the sprint, read the documentation carefully, review your changes in detail before sending them, and run the validation and the rendering of the html to make sure everything is all right. It's a lot of work for us to review if there are so many errors in the PR. Thanks!

**Comment by datapythonista** [comment]
@brute4s99 can you update based on the previous review?

**Comment by datapythonista** [comment]
@brute4s99 do you have time to make the required fixes?

**Comment by brute4s99** [comment]
> @brute4s99 do you have time to make the required fixes?

I am currently working on it, @datapythonista . I am sorry it is taking so much time :/

**Comment by datapythonista** [comment]
Closing in favor of #23936

**Pull request closed without merging**